sky/task.py

```diff
@@ -269,10 +320,12 @@ def from_yaml_config(config: Dict[str, Any]) -> 'Task':
         all_storages = fm_storages
         for storage in all_storages:
             mount_path = storage[0]
-            assert mount_path, \
-                'Storage mount path cannot be empty.'
+            assert mount_path, 'Storage mount path cannot be empty.'
             try:
-                storage_obj = storage_lib.Storage.from_yaml_config(storage[1])
+                storage_config = _fill_in_env_vars_in_storage_config(
+                    storage[1], task.envs)
+                storage_obj = storage_lib.Storage.from_yaml_config(
+                    storage_config)
```

Comment:
Instead of overriding the `storage_config`, should we just replace all the env var in the `file_mounts` section? For example, in L285 we do:
```
if config.get('file_mounts') is not None:
    file_mounts_str = yaml.dumps(config['file_mounts'])
    new_file_mounts_str = _replace_env_var(file_mounts_str, config.get('envs', {}))
    config['file_mounts'] = yaml.loads(new_file_mounts_str)
```

With that, the user can do something like the following:
```
file_mounts:
  /model_path/llama-${SIZE}b: s3://llama-weights/llama-${SIZE}b
```

Reply:
One worry with string replacement is that it'll make some weird things "work":
```
file_mounts:
  /path:
    mo${empty_envvar}de: xxxx
```
I think we should make replacement more structured.

That said, supporting env vars in src/dst seems a useful case. How about we support it when users ask?

Reply:
Sounds good, though I still think the replacement for `dst` and `src` might be quite important, as that was one thing we would like to have for the Vicuna for downloading the correct llama weight from the bucket based on the envvar.

For the problem you mentioned, we can do the schema validation before the replacement happens to avoid that issue.

Reply:
Good call! Updated as such.